TYPO3.Media/Classes/TYPO3/Media/Controller/AssetController.php

```diff
@@ -175,8 +175,13 @@ public function indexAction($view = null, $sort = null, $filter = null, $tagMode
             }
         }
 
-        if ($this->browserState->get('sort') !== 'Modified') {
-            $this->assetRepository->setDefaultOrderings(array('resource.filename' => \TYPO3\Flow\Persistence\QueryInterface::ORDER_ASCENDING));
+        if ($this->browserState->get('sort') !== 'Modified-ASC') {
+            $sortOptions = explode('-', $this->browserState->get('sort'));
+
+            ($sortOptions[0] == 'Name') ? $sortFor = 'resource.filename' : $sortFor = 'lastModified';
+            ($sortOptions[1] == 'DESC') ? $sortDirection = \TYPO3\Flow\Persistence\QueryInterface::ORDER_DESCENDING : $sortDirection = \TYPO3\Flow\Persistence\QueryInterface::ORDER_ASCENDING;
```

Comment:
I'd strongly suggest to split the sorting into two parameters instead. It will make it easier to determine in conditions in the templates and make it more clear.